### PR TITLE
refactor: makefile: reference binaries using variables

### PR DESCRIPTION
**Pull request by miguelsorianod**
## Description

Refactor the Makefile to use makefile variables to reference some of the binaries we are using.
The criteria that has been used to move some of the binaries references to variables is to move those where the binary might not be "by default" installed in the system or it is potentially susceptible to be changed by the user.

## Verification Steps

Makefile targets still work as expected.

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has been created for changes required on the client side

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -873,24 +882,22 @@ deploy/service: deploy/envoy deploy/route
...
-	@-oc process -f ./templates/secrets-template.yml | oc delete -f - -n $(NAMESPACE)
-	@-oc process -f ./templates/route-template.yml | oc delete -f - -n $(NAMESPACE)
-	@-oc delete -f ./templates/envoy-config-configmap.yml -n $(NAMESPACE)
-	@-oc process -f ./templates/service-template.yml \
+	@-$(OC) process -f ./templates/observatorium-token-refresher.yml | oc delete -f - -n $(NAMESPACE)
+	@-$(OC) process -f ./templates/db-template.yml | oc delete -f - -n $(NAMESPACE)
+	@-$(OC) process -f ./templates/secrets-template.yml | oc delete -f - -n $(NAMESPACE)
+	@-$(OC) process -f ./templates/route-template.yml | oc delete -f - -n $(NAMESPACE)
+	@-$(OC) delete -f ./templates/envoy-config-configmap.yml -n $(NAMESPACE)
+	@-$(OC) process -f ./templates/service-template.yml \
```
Comment (VanillaSpoon) [inline Makefile:897]: Hey, good idea :)
Just wondering if we should be adding the references in the later half also? Similar to this:

```diff
-        @-$(OC) process -f ./templates/observatorium-token-refresher.yml | oc delete -f - -n $(NAMESPACE)
+        @-$(OC) process -f ./templates/observatorium-token-refresher.yml | $(OC) delete -f - -n $(NAMESPACE)
```

Comment (miguelsorianod) [reply Makefile:897]: That's correct. I missed those. I will update the contents :+1:

Comment (miguelsorianod) [reply Makefile]: updated

File: Makefile [hunk 2]
```diff
@@ -691,19 +700,19 @@ endif
...
 deploy/project:
-	@-oc new-project $(NAMESPACE)
+	@-$(OC) new-project $(NAMESPACE)
 .PHONY: deploy/project
 
 # deploy the postgres database required by the service to an OpenShift cluster
 deploy/db:
-	oc process -f ./templates/db-template.yml | oc apply -f - -n $(NAMESPACE)
+	$(OC) process -f ./templates/db-template.yml | oc apply -f - -n $(NAMESPACE)
 	@time timeout --foreground 3m bash -c "until oc get pods -n $(NAMESPACE) | grep kas-fleet-manager-db | grep -v deploy | grep -q Running; do echo 'database is not ready yet'; sleep 10; done"
```
Comment (VanillaSpoon) [inline Makefile:709]: So theres a few other instances similar to the previous comment, I'll just mark them out :)
```diff
-	$(OC) process -f ./templates/db-template.yml | oc apply -f - -n $(NAMESPACE)
-	@time timeout --foreground 3m bash -c "until oc get pods -n $(NAMESPACE) | grep kas-fleet-manager-db | grep -v deploy | grep -q Running; do echo 'database is not ready yet'; sleep 10; done"
+	$(OC) process -f ./templates/db-template.yml | $(OC) apply -f - -n $(NAMESPACE)
+	@time timeout --foreground 3m bash -c "until $(OC) get pods -n $(NAMESPACE) | grep kas-fleet-manager-db | grep -v deploy | grep -q Running; do echo 'database is not ready yet'; sleep 10; done"
```

File: Makefile [hunk 3]
```diff
@@ -805,7 +814,7 @@ deploy/service: DYNAMIC_SCALING_CONFIG ?= "{developer: {reserved_streaming_units
 deploy/service: deploy/envoy deploy/route
 	@if test -z "$(IMAGE_TAG)"; then echo "IMAGE_TAG was not specified"; exit 1; fi
 	@time timeout --foreground 3m bash -c "until oc get routes -n $(NAMESPACE) | grep -q kas-fleet-manager; do echo 'waiting for kas-fleet-manager route to be created'; sleep 1; done"
```
Comment (VanillaSpoon) [inline Makefile:816]:
```diff
-	@time timeout --foreground 3m bash -c "until oc get routes -n $(NAMESPACE) | grep -q kas-fleet-manager; do echo 'waiting for kas-fleet-manager route to be created'; sleep 1; done"

+	@time timeout --foreground 3m bash -c "until $(OC) get routes -n $(NAMESPACE) | grep -q kas-fleet-manager; do echo 'waiting for kas-fleet-manager route to be created'; sleep 1; done"
```

**Comment by VanillaSpoon** [comment]
Hey Sorry @miguelsorianod
There were a few similar occurences, but other than than lgtm :)

File: Makefile (hunk 3 above)
Comment (miguelsorianod) [reply Makefile]: updaed

File: Makefile (hunk 2 above)
Comment (miguelsorianod) [reply Makefile]: updated

**Review by VanillaSpoon — APPROVED** [review]

**Pull request merged**